tools/NuGet/template-nuget/DynamoVisualProgramming.Core.nuspec

```diff
@@ -29,19 +32,54 @@
                 <dependency id="Newtonsoft.Json" version="13.0.1"/>
                 <dependency id="System.Collections.Immutable" version="1.7.1"/>
             </group>
+            <!--TODO - why do we reference nunit from dynamo core? This is an odd partial collection of dependencies to mark.-->
+            <group targetFramework="net6.0">
+                <dependency id="NUnit" version="3.13.3"/>
+                <dependency id="DynamoVisualProgramming.ZeroTouchLibrary" version="$Version$"/>
+                <dependency id="Newtonsoft.Json" version="13.0.1"/>
+                <dependency id="System.Collections.Immutable" version="5.0.0"/>
+            </group>
```

Comment:
one strategy I used in the past (to make sure I got the dependencies correct) was to generate packages (and nuspecs) for every package via csproj automation and merging them manually in the final nuspecs.

Reply:
will take a look at this - I am not convinced we actually want a fully transitive list of deps given the use cases we actually provide.

AFAIK these packages are not used for actually shipping any of our binaries, OR our deps - just for compiling against them. In fact, if devs ship these binaries it will cause conflicts when dyn packages/extensions are loaded into Dynamo/Revit etc.

Though - you could make the argument we don't really provide a public nuget for actually integrating and shipping Dynamo into a another project - but that seems to be a different missing package - or require a total rethink of these packages.

Reply:
@pinzart90 do you know if there is a way to provide a default package reference that excludes all binaries from being consumed? Or should we provide some docs on how to reference these packages with that snippet?

Reply:
https://learn.microsoft.com/en-us/nuget/create-packages/creating-a-package#from-a-convention-based-working-directory
You can also control compile/runtime references in the PackageReference syntax

Reply:
If it is just about compiling, I think dotnet6 has a build property that allows a project to produce reference only assemblies (just the public APIs) 
https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#producereferenceassemblyinoutdir
We could include them only in the ref/{tfm} subfolder so that they are only usable at compile time (I have not tested this yet)

Reply:
interesting idea @pinzart90 - I'll try that, it seems less flexible, but less likely to cause issues for clients, and more idiomatic given how we expect most users to actually use these packages. This is a repeated issue I see over and over on the forum etc - where users need to make copy local false on all our binaries.

Maybe the `DVP.Core` package and `DVP.Tests` package could continue using the `lib` folder as `DVP.Core` can be used to integrate Dynamo just the core of Dynamo.... I think.

FYI @twastvedt @QilongTang for feedback on this idea.

Anyhow - what do you think about merging this without the discussed `lib` -> `ref`, and I can tackle that in a separate PR / followup, I think it's a very good idea to test out and get feedback internally and externally from clients - but I'd like to get this merged so we can unblock player, revit etc.

Reply:
Sounds good to me

Reply:
A separate PR to put dlls in `ref` makes sense to me. Not sure about whether they should be reference assemblies or not though. That does seem typical, and would make for somewhat smaller nugets (not critical, I think). However, then consumers would lose the ability to see decompiled source logic when they click "Go to definition" on a Dynamo API, which can be handy.

Reply:
I'm missing something.
1. Why would this make smaller nugets?
2. Why would consumers lose the decompile source option in vs? Is this documented somewhere?

My understanding -which could be incorrect, was that using the `ref` folder just kept the binaries from being copied to the output path of the consuming project.

Reply:
Sorry, yes, confusing. We can manually put any dll we like in the ref folder. But Microsoft also has the concept of a reference assembly. In idiomatic nuget packages (I think) the `ref` folder contains [reference assemblies](https://learn.microsoft.com/en-us/dotnet/standard/assembly/reference-assemblies). These assemblies have the implementation and private properties compiled out.

Reply:
@twastvedt thanks very interesting - I will note that this should be tested when I/we work on this. From my reading- I think that since we are not actually generating reference assemblies, we'll get both "features" that we want - though it won't be completely idiomatic.
1.  users's won't have to manually remove all the dynamo binaries/deps from their output folders
2. the assemblies will still contain implementation.

I think this is a good compromise for most of these packages. Less convinced for dynamo core package.

Anyway, thanks for the info.